### PR TITLE
chore get new loans list

### PR DESCRIPTION
**Pull request by aziyatali**
Removed some redundant codes and added newProduct list based joining it with investments whether current user has invested or not!

### CODE REVIEW

**Comment by Jscripter-pk** [comment]
@aziyatali there is no need of optional user_id to be passed it can be checked via parsing jwt token and checking if token is not expired then use the user_id from there

**Comment by aziyatali** [comment]
@Jscripter-pk So, just adding parserToken function would solve extra attribute right?

**Comment by aziyatali** [comment]
@Jscripter-pk Islam, added token parser in utils and did as you instructed. Please kindly review

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
@aziyatali please don't use the naming directly from newloan. Please give some better names

**Comment by aziyatali** [comment]
@Jscripter-pk Please take a look!

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
Few things are missing 
also check for token expiration
and await for getting parsed token so that it don't lead to inconsistency

**Comment by aziyatali** [comment]
@Jscripter-pk Please review this PR, refactored some tests & skipped post requests since we do not have valid parameters. Let's test them while integrating once we finish loans & loans detail

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
@aziyatali 
1. Instead of skipping test mock them.
 2. I see you remove params from even get functions please make sure that atleast pass some params that make sure that they work as expected.

**Comment by aziyatali** [comment]
@Jscripter-pk Let's create new PR for mocking although we didn't really have tests for post requests from the beginning. Since this PR is for loans joining with investments so we can get started with integration asap

**Comment by Jscripter-pk** [comment]
@aziyatali when you make a PR related to one specific thing make sure you only touch that part as It messes things up for reviews. Let me first just release a beta version of it and see if it works

**Comment by aziyatali** [comment]
Got it!

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
@aziyatali please add back infinite query implementation back that I added.

File: src/kross-client/loans.ts [hunk 1]
```diff
@@ -42,19 +44,47 @@ export class Loans extends KrossClientBase {
...
     });
-
-    return recentProduct;
-  }
+    const loansArray = Object.values(loan?.data);
+    const loansResponseArray = loansArray.map((item: any): LoansResponse => {
+      if (userData?.user_id) {
+        const inv = item.investments.find((invItem: any) => invItem?.userId == userData.user_id);
+        return {
+          ...item,
```
Comment (Jscripter-pk) [inline src/kross-client/loans.ts:59]: and moreover I asked you to not use the names from newloan 
I would prefer isUserInvest over is_user_invest

**Comment by aziyatali** [comment]
> @aziyatali please add back infinite query implementation back that I added.

@Jscripter-pk I didn't touch to LoanData you implemented, what do you mean?

File: src/kross-client/loans.ts (hunk 1 above)
Comment (aziyatali) [reply src/kross-client/loans.ts:59]: Those are the table attributes, checked production table and followed the naming convention there. 

Screen Shot 2023-01-31 at 3 05 34 PM

**Comment by Jscripter-pk** [comment]
@aziyatali I checked what you did in a seperate function that should be done in loans function.
I think at one place JH have told to go with generic one. As Loans screen is using loanData() function so please make it through loanData instead of creating a new function.

**Comment by aziyatali** [comment]
> @aziyatali I checked what you did in a seperate function that should be done in loans function. I think at one place JH have told to go with generic one. As Loans screen is using loanData() function so please make it through loanData instead of creating a new function.

loanData is a hook function which gets loans. They are same.
I know that we won’t be using loanData hook you implemented instead use loan hook I created for loans & loan details page as per our yerderday's discussion.

File: src/kross-client/loans.ts [hunk 2]
```diff
@@ -80,12 +70,36 @@ export class Loans extends KrossClientBase {
...
             });
+            const loansArray = Object.values(loan?.data);
+            const loansResponseArray = loansArray.map(
+              (item: any): LoansResponse => {
+                if (userData?.user_id) {
+                  const inv = item.investments.find(
+                    (invItem: any) => invItem?.userId == userData.user_id
+                  );
+                  return {
+                    ...item,
```
Comment (Jscripter-pk) [inline src/kross-client/loans.ts:88]: @aziyatali overall now it looks fine just change this part as we are passing this from sdk but not from db directly so I think we can use isUserIvest casing here

Comment (aziyatali) [reply src/kross-client/loans.ts:88]: Got it!

**Review by Jscripter-pk — APPROVED** [review]

**Pull request merged**